client/components/tinymce/plugins/wpcom-view/views/embed/view.jsx

```diff
@@ -19,25 +16,12 @@ import QueryEmbed from 'components/data/query-embed';
 import ResizableIframe from 'components/resizable-iframe';
 
 class EmbedView extends Component {
-	state = {
-		wrapper: null,
-	};
+	view = createRef();
+
+	iframe = createRef();
 
 	componentDidMount() {
-		// Rendering the frame follows a specific set of steps, whereby an
-		// initial rendering pass is made, at which time the frame is rendered
-		// in a second pass, before finally setting the frame markup.
-		//
-		// TODO: Investigate and evaluate whether we need to avoid rendering
-		//       the iframe on the initial render pass
-		// eslint-disable-next-line react/no-did-mount-set-state
-		this.setState(
-			{
-				// eslint-disable-line react/no-did-mount-set-state
-				wrapper: this.refs.view,
-			},
-			this.setHtml
-		);
```

Comment:
I'm afraid this `setState()` call, which forces a two pass render, might be necessary. It first renders the container `<div>` and then, in second pass, creates the `<iframe>`. Instead of rendering `<div><iframe></div>` in one pass.

When `iframe` is being created in DOM, it might go through some weird intermediate state where `iframe.contentDocument` or `iframe.contentWindow` are null. And they are initialized only a moment later, when some async process inside the browser finishes.

If `iframe.contentDocument` is `null`, our code can either crash, or fail to set the content HTML.

I've been debugging this a few months ago inside the `components/web-preview` component and still don't understand it.

Here's the relevant definition from the spec: https://html.spec.whatwg.org/multipage/iframe-embed-object.html#dom-iframe-contentwindow

I don't know what "nested browsing context" means and when it exists or not.

Reply:
I'm not sure either 🤷  I prefer closing this one to avoid introducing subtle bugs. Let's revisit in the future 👍